Python/initconfig.c

```diff
@@ -40,99 +57,181 @@ typedef enum {
     PyConfig_MEMBER_WSTR_LIST = 12,
 } PyConfigMemberType;
 
+typedef enum {
+    // Option which cannot get or set by PyConfig_Get() and PyConfig_Set()
+    PyConfig_MEMBER_INIT_ONLY = 0,
+
+    // Option which cannot be set by PyConfig_Set()
+    PyConfig_MEMBER_READ_ONLY = 1,
+
+    // Public option: can be get or set by PyConfig_Get() and PyConfig_Set()
+    PyConfig_MEMBER_PUBLIC = 2,
+} PyConfigMemberVisibility;
+
+typedef struct {
+    const char *attr;
+    int flag_index;
+    config_sys_flag_setter flag_setter;
+} PyConfigSysSpec;
+
 typedef struct {
     const char *name;
     size_t offset;
     PyConfigMemberType type;
+    PyConfigMemberVisibility visibility;
+    PyConfigSysSpec sys;
 } PyConfigSpec;
 
-#define SPEC(MEMBER, TYPE) \
-    {#MEMBER, offsetof(PyConfig, MEMBER), PyConfig_MEMBER_##TYPE}
+#define SPEC(MEMBER, TYPE, visibility, sys) \
+    {#MEMBER, offsetof(PyConfig, MEMBER), \
+     PyConfig_MEMBER_##TYPE, (visibility), sys}
+
+#define SYS_ATTR(name) {name, -1, NULL}
+#define SYS_FLAG_SETTER(index, setter) {NULL, index, setter}
+#define SYS_FLAG(index) SYS_FLAG_SETTER(index, NULL)
+#define NO_SYS SYS_ATTR(NULL)
+
+#define INIT_ONLY PyConfig_MEMBER_INIT_ONLY
+#define READ_ONLY PyConfig_MEMBER_READ_ONLY
+#define PUBLIC PyConfig_MEMBER_PUBLIC
```

Comment:
```suggestion
#define SPEC(MEMBER, TYPE, VISIBILITY, sys) \
    {#MEMBER, offsetof(PyConfig, MEMBER), \
     PyConfig_MEMBER_##TYPE, PyConfig_MEMBER_##VISIBILITY, sys}

#define SYS_ATTR(name) {name, -1, NULL}
#define SYS_FLAG_SETTER(index, setter) {NULL, index, setter}
#define SYS_FLAG(index) SYS_FLAG_SETTER(index, NULL)
#define NO_SYS SYS_ATTR(NULL)
```

This *could* be a bit simpler since you won't need additional macros (don't forget to remove the undef). But if you want to macrofiy the enumeration fields, you can keep the current code.